pipeline/src/pipeline.rs

```diff
@@ -398,6 +398,26 @@ impl<T: FieldElement> Pipeline<T> {
         }
     }
 
+    pub fn from_maybe_optimized_file(self, file: PathBuf) -> Self {
+        if file.extension().unwrap() == "cbor" {
```

Comment:
I think we should use a non-generic file extension, and one that allows being grepped also by `*.pil*`
so either `.pilo` (PIL optimized or PIL object) or `.pilb` (PIL binary).

Reply:
Or if temporaries that people aren't supposed to use `.pilo` (PIL object), `.pila` (PIL artifact), `.pilt` (PIL temporary), `.pili` (PIL intermediate), ...

Reply:
Makes sense, `.pilo` or `.pil.cbor` sounds best to me, but the others are fine too